src/Microsoft.AspNet.Mvc.ViewFeatures/Rendering/Html/DefaultHtmlGenerator.cs

```diff
@@ -1302,12 +1302,26 @@ private IHtmlContent GenerateGroupsAndOptions(string optionLabel, IEnumerable<Se
             foreach (var group in groupedSelectList)
             {
                 var optGroup = group.First().Group;
+                BufferedHtmlContent optGroupContent;
 
-                // Wrap if requested.
-                TagBuilder groupBuilder = null;
                 if (optGroup != null)
                 {
-                    groupBuilder = new TagBuilder("optgroup");
+                    optGroupContent = new BufferedHtmlContent();
+                    optGroupContent.Append(Environment.NewLine);
+                }
+                else
+                {
+                    optGroupContent = listItemBuilder;
```

Comment:
This and [lines 1335,1336](https://github.com/aspnet/Mvc/pull/2952/files#diff-0e83439eeed8ffbde4eceb0db1f6e541R1336) mean the `listItemBuilder` is placed in the `listItemBuilder` content. How does this not stack dive?